### PR TITLE
Change the params of scan_begin which defined in TableAm.

### PR DESCRIPTION
**Pull request by wenchaozhang-123**
Authored-by: Zhang Wenchao zwcpostgres@gmail.com

fix #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

**Review by my-ship-it — APPROVED** [review]
LGTM

**Comment by gongxun0928** [comment]
LGTM
it‘s better to fill in the information in the PR template every time.

**Review by gongxun0928 — APPROVED** [review]
LGTM
it‘s better to fill in the information in the PR template every time.

File: src/include/access/tableam.h [hunk 1]
```diff
@@ -313,12 +313,14 @@ typedef struct TableAmRoutine
...
 	 * needs to be deallocated at scan_end (ScanOptions's SO_TEMP_SNAPSHOT).
+	 *
+	 * `ctx` is a context pointer that can be used to pass information from analyze or other scan types.
 	 */
 	TableScanDesc (*scan_begin) (Relation rel,
 								 Snapshot snapshot,
 								 int nkeys, struct ScanKeyData *key,
 								 ParallelTableScanDesc pscan,
-								 uint32 flags);
+								 uint32 flags, void *ctx);
```
Comment (yjhjstz) [inline src/include/access/tableam.h:323]: how about add new interface `scan_begin _ctx ` ?

Comment (wenchaozhang-123) [reply src/include/access/tableam.h:323]: Hm...  It is also a good solution. In my opinion, I prefer to add a param to existing interface, for the reason the scan_begin_ctx will always not be used in cbdb kernel. Maybe we can ask @my-ship-it to confirm this.

Comment (yjhjstz) [reply src/include/access/tableam.h:323]: hm, `scan_begin ` add ad param will break other extension that use table-am. like citus etc.

Comment (wenchaozhang-123) [reply src/include/access/tableam.h:323]: Ok, that  is also a problem.

**Pull request closed without merging**